crates/nu-protocol/src/value/mod.rs

```diff
@@ -1049,52 +1045,57 @@ impl Value {
                         // Create a List which contains each matching value for contained
                         // records in the source list.
                         Value::List { vals, .. } => {
-                            // TODO: this should stream instead of collecting
-                            let mut output = vec![];
-                            for val in vals {
-                                // only look in records; this avoids unintentionally recursing into deeply nested tables
-                                if matches!(val, Value::Record { .. }) {
-                                    if let Ok(result) = val.clone().follow_cell_path(
-                                        &[PathMember::String {
-                                            val: column_name.clone(),
-                                            span: *origin_span,
-                                            optional: *optional,
-                                        }],
-                                        insensitive,
-                                    ) {
-                                        output.push(result);
-                                    } else {
-                                        return Err(ShellError::CantFindColumn {
-                                            col_name: column_name.to_string(),
+                            let list = vals
+                                .into_iter()
+                                .map(|val| {
+                                    let val_span = val.span();
+                                    // only look in records; this avoids unintentionally recursing into deeply nested tables
```

Comment:
Was this recursing before for a table in a table and now wouldn't?

Reply:
I'm not quite sure what the comment means (looking at it briefly only), but it was there before.

Reply:
I mean that previously in the following body there was a recursion with a newly created cellpath member, and now you implement that yourself. (just to make sure previously it was in fact recursing deeper than intended?)

Reply:
Sorry, I meant that I don't know what the `// only look in records; this avoids unintentionally recursing into deeply nested tables` comment is trying to say (I just copied it over from the old code).

But yes, before there was some recursion (only one level deep) into records, and I've replaced that with the same code above for `Record`. Although, it looks like I've removed/forgotten the `did_you_mean(val.columns(), column_name)` portion.

Reply:
If I read https://github.com/nushell/nushell/pull/11066/commits/b300687a5a6dcffe79ddd343f92897f0b56a9a82 correctly this should be fixed

Reply:
Yep, should be now.